ref-farming/src/farmer.rs

```diff
@@ -52,7 +52,7 @@ impl Farmer {
     pub(crate) fn sub_reward(&mut self, token: &AccountId, amount: Balance) -> Balance {
         let value = *self.rewards.get(token).expect(ERR21_TOKEN_NOT_REG);
         assert!(value >= amount, "{}", ERR22_NOT_ENOUGH_TOKENS);
-        if amount == 0 {
+        if amount == 0 || amount == value {
             self.rewards.remove(&token.clone());
```

Comment:
Don't need to clone here
```suggestion
            self.rewards.remove(&token);
```